omnibus/files/private-chef-cookbooks/private-chef/templates/default/postgresql.conf.erb

```diff
@@ -199,7 +199,7 @@ archive_timeout = <%= node['private_chef']['postgresql']['archive_timeout'] %>
 
 # These settings are ignored on a standby server
 
-#max_wal_senders = 0		# max number of walsender processes
+max_wal_senders = 0		# max number of walsender processes
```

Comment:
could you add some context about why this needs to be enabled?

Reply:
For context, see the commit titled "Set max_wal_senders = 0".  Or, did you mean in a comment?

Reply:
From the commit message:
```
    postgresql12 service would not start.  setting max_wal_senders = 0 in
    postgresql.conf seems to have fixed the issue, at least for a simple
    standalone configuration.  two excerpts from release notes which may
    have affected the issue are included below.

    RELEASE NOTES FROM POSTGRES 10:

    <<Reduce configuration changes necessary to perform streaming backup and replication (Magnus Hagander, Dang Minh Huong)

    Specifically, the defaults were changed for wal_level, max_wal_senders, max_replication_slots, and hot_standby to make them suitable for these usages out-of-the-box.>>

    RELEASE NOTES FROM POSTGRES 12:

    <<Make max_wal_senders not count as part of max_connections>>

    PRIOR STARTUP ERROR:

    Recipe: private-chef::postgresql
      * component_runit_service[postgresql] action restart
      Recipe: <Dynamically Defined Resource>
        * service[postgresql] action nothing (skipped due to action :nothing)
        * runit_service[postgresql] action restart

          ================================================================================
          Error executing action `restart` on resource 'runit_service[postgresql]'
          ================================================================================

          Mixlib::ShellOut::ShellCommandFailed
          ------------------------------------
          Expected process to exit with [0], but received '1'
          ---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
          STDOUT: timeout: down: /opt/opscode/service/postgresql: 0s, normally up, want up
          STDERR:
          ---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
          Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql returned 1

          Cookbook Trace:
          ---------------
          /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:136:in `tap'
          /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:136:in `safe_sv_shellout!'
          /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:164:in `restart_service'
          /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:327:in `block in <class:RunitService>'

          Resource Declaration:
          ---------------------
          # In /var/opt/opscode/local-mode-cache/cookbooks/enterprise/resources/component_runit_service.rb

           83:     runit_service new_resource.component do
           84:       action action_name
           85:     end
           86:   end

          Compiled Resource:
          ------------------
          # Declared in /var/opt/opscode/local-mode-cache/cookbooks/enterprise/resources/component_runit_service.rb:83:in `block (2 levels) in class_from_file'

          runit_service("postgresql") do
            action [:restart]
            default_guard_interpreter :default
            declared_type :runit_service
            cookbook_name "private-chef"
            service_name "postgresql"
            options {}
            run_template_name "postgresql"
            log_dir "/var/log/postgresql"
            log_template_name "postgresql"
            check_script_template_name "postgresql"
            finish_script_template_name "postgresql"
            control_template_names {}
            status_command "/opt/opscode/embedded/bin/sv status /opt/opscode/service"
          end

          System Info:
          ------------
          chef_version=15.15.1
          platform=ubuntu
          platform_version=18.04
          ruby=ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-linux]
          program_name=/opt/opscode/embedded/bin/chef-client
          executable=/opt/opscode/embedded/bin/chef-client

        ================================================================================
        Error executing action `restart` on resource 'component_runit_service[postgresql]'
        ================================================================================

        Mixlib::ShellOut::ShellCommandFailed
        ------------------------------------
        runit_service[postgresql] (/var/opt/opscode/local-mode-cache/cookbooks/enterprise/resources/component_runit_service.rb line 83) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
        ---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
        STDOUT: timeout: down: /opt/opscode/service/postgresql: 0s, normally up, want up
        STDERR:
        ---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
        Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql returned 1

        Cookbook Trace:
        ---------------
        /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:136:in `tap'
        /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:136:in `safe_sv_shellout!'
        /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:164:in `restart_service'
        /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:327:in `block in <class:RunitService>'

        Resource Declaration:
        ---------------------
        # In /var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/postgresql.rb

         67: component_runit_service 'postgresql' do
         68:   control ['t']
         69: end
         70:

        Compiled Resource:
        ------------------
        # Declared in /var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/postgresql.rb:67:in `from_file'

        component_runit_service("postgresql") do
          action [:enable]
          updated true
          default_guard_interpreter :default
          declared_type :component_runit_service
          cookbook_name "private-chef"
          recipe_name "postgresql"
          control ["t"]
          component "postgresql"
        end

        System Info:
        ------------
        chef_version=15.15.1
        platform=ubuntu
        platform_version=18.04
        ruby=ruby 2.6.6p146 (2020-03-31 revision 67876) [x86_64-linux]
        program_name=/opt/opscode/embedded/bin/chef-client
        executable=/opt/opscode/embedded/bin/chef-client

    Recipe: private-chef::oc_bifrost
      * component_runit_service[oc_bifrost] action restart
      Recipe: <Dynamically Defined Resource>
        * service[oc_bifrost] action nothing (skipped due to action :nothing)
        * runit_service[oc_bifrost] action restart (up to date)
         (up to date)
    Recipe: private-chef::oc_id
      * component_runit_service[oc_id] action restart
      Recipe: <Dynamically Defined Resource>
        * service[oc_id] action nothing (skipped due to action :nothing)
        * runit_service[oc_id] action restart (up to date)
         (up to date)
    Recipe: private-chef::nginx
      * component_runit_service[nginx] action restart
      Recipe: <Dynamically Defined Resource>
        * service[nginx] action nothing (skipped due to action :nothing)
        * runit_service[nginx] action restart (up to date)
         (up to date)
    Recipe: private-chef::elasticsearch
      * component_runit_service[elasticsearch] action restart
      Recipe: <Dynamically Defined Resource>
        * service[elasticsearch] action nothing (skipped due to action :nothing)
        * runit_service[elasticsearch] action restart (up to date)
         (up to date)
    Recipe: private-chef::bookshelf
      * component_runit_service[bookshelf] action restart
      Recipe: <Dynamically Defined Resource>
        * service[bookshelf] action nothing (skipped due to action :nothing)
        * runit_service[bookshelf] action restart (up to date)
         (up to date)
    Recipe: private-chef::opscode-erchef
      * component_runit_service[opscode-erchef] action restart
      Recipe: <Dynamically Defined Resource>
        * service[opscode-erchef] action nothing (skipped due to action :nothing)
        * runit_service[opscode-erchef] action restart (up to date)
         (up to date)
    Recipe: private-chef::partybus
      * execute[set initial migration level] action run
        - execute cd /opt/opscode/embedded/service/partybus && ./bin/partybus init
      * ruby_block[migration-level file sanity check] action run (skipped due to not_if)

    Running handlers:
    Running handlers complete
    Chef Infra Client failed. 405 resources updated in 01 minutes 56 seconds
    [2021-02-12T16:41:19+00:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
    [2021-02-12T16:41:19+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
    [2021-02-12T16:41:19+00:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: component_runit_service[postgresql] (private-chef::postgresql line 67) had an error: Mixlib::ShellOut::ShellCommandFailed: runit_service[postgresql] (/var/opt/opscode/local-mode-cache/cookbooks/enterprise/resources/component_runit_service.rb line 83) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
    ---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
    STDOUT: timeout: down: /opt/opscode/service/postgresql: 0s, normally up, want up
    STDERR:
    ---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql ----
    Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/postgresql returned 1
    <F0><9F><9A><A8> Error: The command exited with status 1
    Running plugin omnibus post-command hook
    Running global pre-exit hook
    Running plugin omnibus pre-exit hook
```